rpcs3/Emu/Cell/Modules/cellAudioOut.cpp

```diff
@@ -361,35 +346,22 @@ error_code cellAudioOutConfigure(u32 audioOut, vm::ptr<CellAudioOutConfiguration
 		audio_out_configuration::audio_out& out = cfg.out.at(audioOut);
 
 		// Apparently the set config does not necessarily have to exist in the list of sound modes.
-		// For example 8 channels are used if the downMixer is set even if the PS3 only supports 2 channel output.
 
 		if (out.channels != config->channel || out.encoder != config->encoder || out.downmixer != config->downMixer)
 		{
 			out.channels = config->channel;
 			out.encoder = config->encoder;
-			out.downmixer = config->downMixer;
 
-			// Try to find the best sound mode for this configuration
-			const auto [channels, downmixer] = out.get_channel_count_and_downmixer();
-			const auto it = std::find_if(out.sound_modes.cbegin(), out.sound_modes.cend(), [channels = channels, &out](const CellAudioOutSoundMode& mode)
+			if (config->downMixer <= CELL_AUDIO_OUT_DOWNMIXER_TYPE_B) // Only change downmixer if it's valid
 			{
-				if (mode.type != out.encoder)
-				{
-					return false;
-				}
-
-				if (out.downmixer == CELL_AUDIO_OUT_DOWNMIXER_TYPE_A)
-				{
-					return mode.channel == CELL_AUDIO_OUT_CHNUM_2;
-				}
+				out.downmixer = config->downMixer;
+			}
 
-				if (out.downmixer == CELL_AUDIO_OUT_DOWNMIXER_TYPE_B)
+			// Try to find the best sound mode for this configuration
+			const auto it = std::find_if(out.sound_modes.cbegin(), out.sound_modes.cend(), [&out](const CellAudioOutSoundMode& mode)
 				{
-					return mode.channel == CELL_AUDIO_OUT_CHNUM_6;
-				}
-
-				return mode.channel == static_cast<u8>(channels);
-			});
+					return mode.type == out.encoder && mode.channel == out.channels;
+				});
```

Comment:
this makes no sense

Reply:
Ignoring the donwmix mode in the config matches PS3 behavior, it will still configure the output format according to channel and encoder even if downmix is > 2.
Calling get_channel_count_and_downmixer() doesn't work because I had to change it to return sound_mode.channel instead of the config set by the game, since the config can be invalid and that would cause exceptions.